File: download-images.php
<?php
// Image downloader script for Rocco website
$images = [
    'hero-bg.jpg' => 'https://www.rococo.net.au/wp-content/uploads/2023/05/rococo-interior-1.jpg',
    'menu-hero.jpg' => 'https://www.rococo.net.au/wp-content/uploads/2023/05/rococo-hero-1.jpg',
    'contact-hero.jpg' => 'https://www.rococo.net.au/wp-content/uploads/2023/05/rococo-interior-2.jpg',
    'functions-hero.jpg' => 'https://www.rococo.net.au/wp-content/uploads/2023/05/rococo-interior-3.jpg',
    'story-hero.jpg' => 'https://www.rococo.net.au/wp-content/uploads/2023/05/rococo-hero-2.jpg',
    'pasta-1.jpg' => 'https://www.rococo.net.au/wp-content/uploads/2023/05/rococo-pasta-1.jpg',
    'antipasti-1.jpg' => 'https://www.rococo.net.au/wp-content/uploads/2023/05/rococo-antipasti-1.jpg',
    'dessert-1.jpg' => 'https://www.rococo.net.au/wp-content/uploads/2023/05/rococo-dessert-1.jpg',
    'main-1.jpg' => 'https://www.rococo.net.au/wp-content/uploads/2023/05/rococo-main-1.jpg',
];

// Create images directory if it doesn't exist
if (!file_exists('assets/images')) {
    mkdir('assets/images', 0777, true);
}

foreach ($images as $filename => $url) {
    $path = 'assets/images/' . $filename;
    
    if (!file_exists($path)) {
        $imageData = @file_get_contents($url);
        if ($imageData) {
            file_put_contents($path, $imageData);
            echo "Downloaded: $filename<br>";
        } else {
            echo "Failed to download: $filename<br>";
        }
    } else {
        echo "Already exists: $filename<br>";
    }
}

echo "Image download process complete!";
?>

File: download_images.py
import os
import requests
from bs4 import BeautifulSoup
from urllib.parse import urljoin, urlparse

# URL of the site to scrape images from
SITE_URL = "https://www.rococo.net.au/"
IMG_DIR = "assets/images/"

# Ensure the images directory exists
os.makedirs(IMG_DIR, exist_ok=True)

# Fetch the page
response = requests.get(SITE_URL)
soup = BeautifulSoup(response.text, "html.parser")

# Find all image tags
img_tags = soup.find_all("img")

for img in img_tags:
    img_url = img.get("src")
    if not img_url:
        continue
    # Handle relative URLs
    img_url = urljoin(SITE_URL, img_url)
    # Get image filename
    filename = os.path.basename(urlparse(img_url).path)
    if not filename:
        continue
    # Download image
    try:
        img_data = requests.get(img_url).content
        with open(os.path.join(IMG_DIR, filename), "wb") as f:
            f.write(img_data)
        print(f"Downloaded {filename}")
    except Exception as e:
        print(f"Failed to download {img_url}: {e}")

print("Image download complete.")
